src/vos/vos_aggregate.c

```diff
@@ -621,8 +626,12 @@ prepare_segments(struct agg_merge_window *mw)
 	 * segments (at most mw_lgc_cnt) and truncated segments (at most
 	 * mw_phy_cnt).
 	 */
-	D_ASSERT(mw->mw_lgc_cnt > 0);
+	D_ASSERT(mw->mw_lgc_cnt > 0 || mw->mw_rmv_cnt > 0);
 	D_ASSERT(mw->mw_phy_cnt > 0);
```

Comment:
Seems current code doesn't guarantee this now, what if a window has delete records only? And I see D_ASSERT(mw->mw_phy_cnt > 0) in other places of window flush code path.

Reply:
I don't think it's possible to have a removal record only because they always correspond to physical records (.e.g. they are only inserted if there is a physical record at the same major epoch.)

Reply:
Ok, I see. I was thinking we could insert removal record larger than existing physical record.